### ISSUE TITLE
Error uploading database

### ISSUE THREAD

**Issue by waltersf** [eubr-bigsea/limonero#35]
This error is thrown but limonero reports success.

```
limonero_1     | 2018-03-23 00:39:19.376451, p15, th140332007921408, ERROR Handle Exception: NamenodeImpl.cpp: 253: File /tmp/upload/1886-citiescsv/cities.csv.part000000
001 could only be replicated to 0 nodes instead of minReplication (=1).  There are 4 datanode(s) running and 4 node(s) are excluded in this operation.
limonero_1     |        at org.apache.hadoop.hdfs.server.blockmanagement.BlockManager.chooseTarget4NewBlock(BlockManager.java:1559)
limonero_1     |        at org.apache.hadoop.hdfs.server.namenode.FSNamesystem.getAdditionalBlock(FSNamesystem.java:3245)
limonero_1     |        at org.apache.hadoop.hdfs.server.namenode.NameNodeRpcServer.addBlock(NameNodeRpcServer.java:663)
limonero_1     |        at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolServerSideTranslatorPB.addBlock(ClientNamenodeProtocolServerSideTranslatorPB.java:482)
limonero_1     |        at org.apache.hadoop.hdfs.protocol.proto.ClientNamenodeProtocolProtos$ClientNamenodeProtocol$2.callBlockingMethod(ClientNamenodeProtocolProtos.ja
va)
limonero_1     |        at org.apache.hadoop.ipc.ProtobufRpcEngine$Server$ProtoBufRpcInvoker.call(ProtobufRpcEngine.java:619)
limonero_1     |        at org.apache.hadoop.ipc.RPC$Server.call(RPC.java:975)
limonero_1     |        at org.apache.hadoop.ipc.Server$Handler$1.run(Server.java:2040)
limonero_1     |        at org.apache.hadoop.ipc.Server$Handler$1.run(Server.java:2036)
limonero_1     |        at java.security.AccessController.doPrivileged(Native Method)
limonero_1     |        at javax.security.auth.Subject.doAs(Subject.java:415)
limonero_1     |        at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1656)
limonero_1     |        at org.apache.hadoop.ipc.Server$Handler.run(Server.java:2034)
limonero_1     | 
limonero_1     |        @       Hdfs::Internal::UnWrapper<Hdfs::UnresolvedLinkException, Hdfs::HdfsIOException, Hdfs::Internal::Nothing, Hdfs::Internal::Nothing, Hdfs::I
nternal::Nothing, Hdfs::Internal::Nothing, Hdfs::Internal::Nothing, Hdfs::Internal::Nothing, Hdfs::Internal::Nothing, Hdfs::Internal::Nothing, Hdfs::Internal::Nothing>::
unwrap(char const*, int)
limonero_1     |        @       Hdfs::Internal::UnWrapper<Hdfs::NotReplicatedYetException, Hdfs::UnresolvedLinkException, Hdfs::HdfsIOException, Hdfs::Internal::Nothing,
 Hdfs::Internal::Nothing, Hdfs::Internal::Nothing, Hdfs::Internal::Nothing, Hdfs::Internal::Nothing, Hdfs::Internal::Nothing, Hdfs::Internal::Nothing, Hdfs::Internal::No
thing>::unwrap(char const*, int)
limonero_1     |        @       Hdfs::Internal::UnWrapper<Hdfs::FileNotFoundException, Hdfs::NotReplicatedYetException, Hdfs::UnresolvedLinkException, Hdfs::HdfsIOExcept
ion, Hdfs::Internal::Nothing, Hdfs::Internal::Nothing, Hdfs::Internal::Nothing, Hdfs::Internal::Nothing, Hdfs::Internal::Nothing, Hdfs::Internal::Nothing, Hdfs::Internal
::Nothing>::unwrap(char const*, int)
limonero_1     |        @       Hdfs::Internal::NamenodeImpl::addBlock(std::string const&, std::string const&, Hdfs::Internal::ExtendedBlock const*, std::vector<Hdfs::In
ternal::DatanodeInfo, std::allocator<Hdfs::Internal::DatanodeInfo> > const&)
limonero_1     |        @       Hdfs::Internal::NamenodeProxy::addBlock(std::string const&, std::string const&, Hdfs::Internal::ExtendedBlock const*, std::vector<Hdfs::I
nternal::DatanodeInfo, std::allocator<Hdfs::Internal::DatanodeInfo> > const&)
limonero_1     |        @       Hdfs::Internal::FileSystemImpl::addBlock(std::string const&, Hdfs::Internal::ExtendedBlock const*, std::vector<Hdfs::Internal::DatanodeIn
fo, std::allocator<Hdfs::Internal::DatanodeInfo> > const&)
limonero_1     |        @       Hdfs::Internal::PipelineImpl::locateNextBlock(std::vector<Hdfs::Internal::DatanodeInfo, std::allocator<Hdfs::Internal::DatanodeInfo> > co
nst&)
limonero_1     |        @       Hdfs::Internal::PipelineImpl::buildForNewBlock()
limonero_1     |        @       Hdfs::Internal::PipelineImpl::PipelineImpl(bool, char const*, Hdfs::Internal::SessionConfig const&, std::shared_ptr<Hdfs::Internal::FileS
ystemInter>, int, int, int, long, Hdfs::Internal::PacketPool&, std::shared_ptr<Hdfs::Internal::LocatedBlock>)
limonero_1     |        @       Hdfs::Internal::OutputStreamImpl::setupPipeline()
```

**Comment by waltersf** [comment]
I downgraded it to tag v1.0 (uses JVM + py4j) and it worked. In HDFS, the logs show the same message. Please, use Lemonade dev environment while testing.

**Comment by michelboaventura** [comment]
I will take a look tomorrow morning. It should not give this error or only report it to the user?

**Comment by waltersf** [comment]
It should not. This message could be related to namenode or datanode problems, but I checked and it seems to be working. I did 3 tests: using command line (provided by HDFS), limonero tag v1.0, and limonero  latest. Only latest raises the error. I checked the configuration property `dfs.client.use.datanode.hostname: true`, but It could be another option that is causing the problem. Try to connect to spark01.<you know the domain>:9000.
